Jenkinsfile

```diff
@@ -102,48 +165,68 @@ pipeline {
 
     stage("Run-Pace-Python-Tests") {
       environment {
-        LD_LIBRARY_PATH = "/usr/local/MATLAB/MATLAB_Runtime/v98/runtime/glnxa64:/usr/local/MATLAB/MATLAB_Runtime/v98/sys/os/glnxa64:/usr/local/MATLAB/MATLAB_Runtime/v98/bin/glnxa64:/usr/local/MATLAB/MATLAB_Runtime/v98/extern/bin/glnxa64"
-        LD_PRELOAD = "/usr/local/MATLAB/MATLAB_Runtime/v98/sys/os/glnxa64/libiomp5.so"
+        LD_LIBRARY_PATH = "/opt/modules-common/software/MATLAB/R${MATLAB_VERSION}/runtime/glnxa64:/opt/modules-common/software/MATLAB/R${MATLAB_VERSION}/bin/glnxa64"
+        LD_PRELOAD = "/opt/modules-common/software/MATLAB/R${MATLAB_VERSION}/sys/os/glnxa64/libiomp5.so"
       }
       steps {
         script {
           if (isUnix()) {
             sh '''
+                module purge
+                module load conda
+                module load matlab/\$MATLAB_VERSION
                 eval "$(/opt/conda/bin/conda shell.bash hook)"
-                conda env remove -n py37
-                conda create -n py37 -c conda-forge python=3.7 -y
-                conda activate py37
-                conda install -c conda-forge scipy euphonic -y
+                conda env remove -n \$ENV_NAME
+                conda create -n \$ENV_NAME -c conda-forge python=\$PYTHON_VERSION -y
+                conda activate \$ENV_NAME
+                pip install numpy scipy euphonic --no-input
+                export MKL_NUM_THREADS=1
                 python -m pip install brille
-                python -m pip install $(find wheelhouse -name "*cp37*whl"|tail -n1)
-                python test/run_test.py || true
+                python -m pip install $(find dist -name "*whl"|tail -n1)
+                timeout --signal 15 6m python test/run_test.py -v
                 test -f success
             '''
           }
           else {
-            powershell './cmake/run_pace_python_tests.ps1'
+            powershell(script:'''
+                conda env remove --prefix ./\$env:ENV_NAME
+                conda create --prefix ./\$env:ENV_NAME -c conda-forge python=\$env:PYTHON_VERSION -y
+                Import-Module "C:/ProgramData/miniconda3/shell/condabin/Conda.psm1"
+                Enter-CondaEnvironment ./\$env:ENV_NAME
+                conda install -c conda-forge scipy euphonic -y
+                python -m pip install brille
+                python -m pip install "dist/$(Get-ChildItem 'dist/*.whl' -name)"
+                python test/run_test.py -v
+            ''', label: "Setup and run tests")
           }
         }
       }
     }
 
     stage("Push release") {
+
       environment {
         GITHUB_TOKEN = get_github_token()
       }
       steps {
         script {
-          if (env.ref_type == 'tag') {
+
+          if (env.BRANCH.startsWith("refs/tags")) {
```

Comment:
Could move this to a `stage-when` block.

Reply:
I don't think I have encountered this before and I currently do not have the time to be trying to get new jenkins commands to run. Will this make much difference?

Reply:
No, not at all, it just skips the stage at an earlier phase in the calculation

Reply:
Considering this is on the 'push release' stage I think we can get away with not making this change for now. The build stage is the slow one. This one takes barely anytime anyway in comparison.